### ISSUE TITLE
NGEN_ASM support is incomplete

### ISSUE THREAD

**Issue by nwnk** [oneapi-src/oneDNN#2061]
It looks like it simply never gets built. `ngen_asm.hpp` is the only place that defines `NGEN_ASM`, and the only place that includes it looks like:
```
#ifdef NGEN_ASM
#include "ngen_asm.hpp"
NGEN_REGISTER_DECL(NGEN_NAMESPACE::AsmCodeGenerator, /* nothing */)
#endif
```
Simply defining the macro is not enough, because unlike the per-gen generators there is no `ELFCodeGenerator` that derives from `AsmCodeGenerator`.

**Comment by densamoilov** [comment]
+ @petercad

**Comment by petercad** [comment]
@nwnk You're right that `ngen_asm.hpp` isn't used within oneDNN, and actually, it may be deleted. Like the rest of nGEN it's a header-only library and is only built as needed.

**Comment by nwnk** [comment]
@petercad 

> @nwnk You're right that `ngen_asm.hpp` isn't used within oneDNN, and actually, it may be deleted. Like the rest of nGEN it's a header-only library and is only built as needed.

Are we sure? [The nGEN README](https://github.com/oneapi-src/oneDNN/tree/main/src/gpu/intel/jit#readme) seems to mention assembly generation as a feature. I would have guessed that it was there for a reason like faster kernel compile times. If that doesn't provide any benefit over CLC then I'm all too happy to remove it, I just want to verify that it's truly unwanted.

**Comment by petercad** [comment]
@nwnk Yes, nGEN is a library used internally in oneDNN, but not all parts of it are used in oneDNN. I verified that oneDNN builds fine without `ngen_asm.hpp`.

**Comment by vpirogov** [comment]
Fixed by #2078.
